Tests/AnalyticsTests/Tests.swift

```diff
@@ -18,9 +18,30 @@ final class Tests: XCTestCase {
         XCTAssertEqual(observer.eventName, ViewEvent.view.name)
         XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
     }
+    
+    func testLogEventAction() {
+        var values = AnalyticsValues()
+        values[keyPath: \.source] = .contactList
+        var action = AnalyticsAction(values: values)
+
+        action(.view)
+        XCTAssertEqual(observer.eventName, ViewEvent.view.name)
+        XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
+    }
+    
+    func testLogEventReplacing() {
+        var values = AnalyticsValues()
+        values[keyPath: \.source] = .contactList
+        var action = AnalyticsAction(values: values)
+
+        action(.view, appending: values)
+        action(.view, replacing: nil)
+        XCTAssertEqual(observer.eventName, ViewEvent.view.name)
+        XCTAssertEqual(observer.params, [:])
```

Comment:
```suggestion
        XCTAssertEqual(observer.params, [:])
        
        action(.view)
        XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
```

This just adds the extra assertion that the original values have not been removed or lost 👍